proto/redpanda/api/dataplane/v1alpha2/cloud_storage.proto

```diff
@@ -0,0 +1,393 @@
+    // Planned: The mount task has been created and is awaiting further actions.
+    STATE_PLANNED = 1;
+    // Preparing: The mount task is gathering resources and preparing for execution.
+    STATE_PREPARING = 2;
+    // Prepared: All preparations are complete, and the mount task is ready to be executed.
+    STATE_PREPARED = 3;
+    // Executing: The mount task is actively transferring or transforming data.
+    STATE_EXECUTING = 4;
+    // Executed: The core mount task actions are complete, but the mount task has not yet cut over or finalized.
+    STATE_EXECUTED = 5;
+    // Cut Over: The mount task has reached a critical point where ownership is transferred or final adjustments are made.
+    STATE_CUT_OVER = 6;
+    // Finished: The mount task has been successfully completed, and no further actions are required.
+    STATE_FINISHED = 7;
+    // Canceling: The mount task is in the process of being canceled, and rollback or cleanup actions may be in progress.
+    STATE_CANCELING = 8;
+    // Cancelled: The mount task has been fully canceled, and no further actions will be taken.
+    STATE_CANCELLED = 9;
+  }
+
+  // Id is the unique identifier for this mount task.
+  int32 id = 1;
+  // State describes the current state of the mount task (e.g. "cancelled").
+  State state = 2;
+  // Type describes the type of this task (mount or unmount).
+  Type type = 3;
+  // Topics is the list of topics that are being mounted or unmounted.
+  repeated Topic topics = 4;
+}
+
+message ListMountTasksRequest {}
+
+message ListMountTasksResponse {
+  repeated MountTask tasks = 1;
+}
+
+message GetMountTaskRequest {
+  // ID is the unique identifier of the mount task to retrieve.
+  int32 id = 1 [
+    // We won't add validation required here, because 0 may be a valid id.
+    (google.api.field_behavior) = REQUIRED
+  ];
+}
+
+message GetMountTaskResponse {
+  MountTask task = 1;
+}
+
+message DeleteMountTaskRequest {
+  // ID is the unique identifier of the mount task to delete.
+  int32 id = 1 [(google.api.field_behavior) = REQUIRED /* We won't add validation required here, because 0 may be a valid id. */];
+}
+
+message DeleteMountTaskResponse {}
+
+message UpdateMountTaskRequest {
+  enum Action {
+    ACTION_UNSPECIFIED = 0;
+    ACTION_PREPARE = 1;
+    ACTION_EXECUTE = 2;
+    ACTION_FINISH = 3;
+    ACTION_CANCEL = 4;
+  }
+
+  // ID is the unique identifier of the mount task to delete.
+  int32 id = 1 [(google.api.field_behavior) = REQUIRED /* We won't add validation required here, because 0 may be a valid id. */];
+
+  // Action is the action to execute on mount task.
+  Action action = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum = {
+      defined_only: true
+      not_in: [0] /* Disallow: Unspecified */
+    }
+  ];
+}
+
+message UpdateMountTaskResponse {}
+
+service CloudStorageService {
+  rpc MountTopics(MountTopicsRequest) returns (MountTopicsResponse) {
```

Comment:
Could we add a description for CloudStorageService? Something along the lines of "Mount and unmount topics in Redpanda clusters. Requires that you have Tiered Storage enabled." This is the description we're using for the Admin API reference - we can also add links to docs when they're ready for Redpanda Cloud.